pkg/kvstore/allocator/allocator.go

```diff
@@ -134,501 +63,109 @@ type Allocator struct {
 	// being derived from the basePrefix.
 	valuePrefix string
 
-	// slaveKeysMutex protects the concurrent access of the slave key by this
-	// agent.
-	slaveKeysMutex lock.Mutex
-
 	// lockPrefix is the prefix to use for all kvstore locks. This prefix
 	// is different from the idPrefix and valuePrefix to simplify watching
 	// for ID and key changes.
 	lockPrefix string
 
-	// min is the lower limit when allocating IDs. The allocator will never
-	// allocate an ID lesser than this value.
-	min idpool.ID
-
-	// max is the upper limit when allocating IDs. The allocator will never
-	// allocate an ID greater than this value.
-	max idpool.ID
-
-	// prefixMask if set, will be ORed to all selected IDs prior to
-	// allocation
-	prefixMask idpool.ID
-
-	// localKeys contains all keys including their reference count for keys
-	// which have been allocated and are in local use
-	localKeys *localKeys
-
 	// suffix is the suffix attached to keys which must be node specific,
 	// this is typical set to the node's IP address
 	suffix string
 
-	// lockless is true if allocation can be done lockless. This depends on
-	// the underlying kvstore backend
-	lockless bool
-
-	// backoffTemplate is the backoff configuration while allocating
-	backoffTemplate backoff.Exponential
-
-	// mainCache is the main cache, representing the allocator contents of
-	// the primary kvstore connection
-	mainCache cache
+	// deleteInvalidPrefixes enables deletion of identities outside of the
+	// valid prefix
+	deleteInvalidPrefixes bool
```

Comment:
To other reviewers - this was moved here because the `mainCache` field is now in the allocator, not in the kvstore backend.